Makefile

```diff
@@ -29,7 +29,9 @@ lint: $(TOOLS_DIR)/golangci-lint
 
 .PHONY: build-container
 build-container:
-	docker build -t $(WEBHOOK_DOCKER_IMAGE_NAME):$(WEBHOOK_DOCKER_IMAGE_TAG) .
+	grep -e "image: $(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" deploy/newrelic-metadata-injection.yaml > /dev/null || \
+	( echo "Docker image tag being built $(DOCKER_IMAGE_TAG) is not synchronized with deployment yaml" && exit 1 )
+	docker build -t $(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG) .
```

Comment:
I spent 30 minutes debugging a timeout of the e2e test because the tag being built was not the same being tested. This might help others get that faster, and will prevent broken releases :)